File: src/main/java/samples/ec01/utils/Consts.java
/*
 * Copyright (C) 2018 INFORMATION SERVICES INTERNATIONAL - DENTSU, LTD. All Rights Reserved.
 * 
 * Unless you have purchased a commercial license,
 * the following license terms apply:
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 * 
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package samples.ec01.utils;

public class Consts {

	public static final String LANGUAGE_JA = "ja";
	public static final String LANGUAGE_EN = "en";
	// Command path
	public static final String LOGIN_COMMAND_PATH = "gem/auth/LoginCommand";
	// Login redirect path
	public static final String LOGIN_REDIRECT_PATH_ATTRIBUTE = "samples.ec01.login.redirect";
	// 登録会員用のグループ
	public static final String PUB_GROUP_CODE = "samples/ec01/Pub001Group";

	// Error message
	public static final String BLANK_MES = "{samples.ec01.bean.validation.isBlank}";
	public static final String NUM_MES = "{samples.ec01.bean.validation.notNum}";
	public static final String NUM_OUT_MES = "{samples.ec01.bean.validation.numOutOfRange}";
	public static final String MAIL_MES = "{samples.ec01.bean.validation.notMail}";
	public static final String KANA_MES = "{samples.ec01.bean.validation.notKana}";
	public static final String USERID_MES1 = "{samples.ec01.bean.validation.UserId.outOfLength}";
	public static final String USERID_MES2 = "{samples.ec01.bean.validation.UserId.invalidChar}";
	public static final String TELL_MES = "{samples.ec01.bean.validation.Tel.invalidChar}";

	// Value format
	public static final int USER_ID_MIN_LENGTH = 4;
	public static final int USER_ID_MAX_LENGTH = 128;
	public static final String USER_ID_FORMART = "^[0-9a-zA-Z\\-@\\_.]+$";
	public static final String TEL_FORMAT = "^[0-9\\-]+$";
	public static final String KANA_FORMAT = "^[ァ-ン]+$";

	// Session scope key
	public static final String CART_BEAN_SESSION_KEY = "cartBean";
	public static final String USER_BEAN_SESSION_KEY = "userBean";
	public static final String SHIPPING_BEAN_SESSION_KEY = "shippingBean";

	// ページでの最大表示件数
	public static final int PAGE_SHOWN_LIMIT = 9;
	public static final String PAGINATION_REQUEST_KEY = "pagination";
	
	// 画像バイナリタイプ
	public static final String BIN_TYPE_SHOP_IMG = "shopImg";
	public static final String BIN_TYPE_PRODUCT_IMG = "productImg";
}

File: src/main/vue/mixins/Consts.js
/*
 * Copyright (C) 2018 DENTSU SOKEN INC. All Rights Reserved.
 * 
 * Unless you have purchased a commercial license,
 * the following license terms apply:
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 * 
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

export const Consts = {
    data: function () {
        return {
            tcPath: tcPath,
            path: {
                productImg: tcPath + '/samples/ec01/resource/bin?type=productImg&id=',
                category: '/product/category?categoryId=',
                newInfo: '/news/newInfo?page='
            },
            apiPath: {
                defaultLayout: tcPath + '/api/samples/ec01/layout/defaultLayout',
                shippingLayout: tcPath + '/api/samples/ec01/layout/shippingLayout',
                top: tcPath + '/api/samples/ec01/top',
                cartInfo: tcPath + '/api/samples/ec01/cart/cartInfo',
                inputCartInfo: tcPath + '/api/samples/ec01/cart/inputCartInfo',
                updateCartInfo: tcPath + '/api/samples/ec01/cart/updateCartInfo',
                deleteCartInfo: tcPath + '/api/samples/ec01/cart/deleteCartInfo?deleteId=',
                doInquiry: tcPath + '/api/samples/ec01/inquiry/doInquiry',
                doLogin: tcPath + '/api/samples/ec01/member/doLogin',
                inputMemberInfo: tcPath + '/api/samples/ec01/member/inputMemberInfo',
                confirmMemberInfo: tcPath + '/api/samples/ec01/member/confirmMemberInfo',
                registMemberInfo: tcPath + '/api/samples/ec01/member/registMemberInfo',
                newInfo: tcPath + '/api/samples/ec01/news/newInfo?page=',
                category: tcPath + '/api/samples/ec01/product/category?categoryId=',
                detail: tcPath + '/api/samples/ec01/product/detail?productId=',
                fulltextSearch: tcPath + '/api/samples/ec01/search/fulltextSearch',
                doSearch: tcPath + '/api/samples/ec01/search/doSearch',
                inputShippingInfo: tcPath + '/api/samples/ec01/shipping/inputShippingInfo',
                inputShippingInfoNoMember: tcPath + '/api/samples/ec01/shipping/inputShippingInfoNoMember',
                confirmShippingInfo: tcPath + '/api/samples/ec01/shipping/confirmShippingInfo',
                doOrder: tcPath + '/api/samples/ec01/shipping/doOrder',
                tradeLaw: tcPath + '/api/samples/ec01/shop/tradeLaw',
                token: tcPath + '/api/samples/ec01/token/outputToken'
            }
        };
    },
    methods: {
        imgUrl: function(productImg) {
            if (this.tcPath === undefined || productImg === undefined) {
                return "";
            } else {
                return this.path.productImg + productImg.lobId;
            }
        },
        categoryUrl: function(categoryId) {
            return this.path.category + categoryId + '&page=';
        },
        newInfoUrl: function(){
            return this.path.newInfo;
        },
        apiDefaultLayout: function() {
            return this.apiPath.defaultLayout;
        },
        apiShippingLayout: function() {
            return this.apiPath.shippingLayout;
        },
        apiTop: function() {
            return this.apiPath.top;
        },
        apiCartInfo: function() {
            return this.apiPath.cartInfo;
        },
        apiInputCartInfo: function(productId) {
            return this.apiPath.inputCartInfo;
        },
        apiUpdateCartInfo: function() {
            return this.apiPath.updateCartInfo;
        },
        apiDeleteCartInfo: function(productId) {
            return this.apiPath.deleteCartInfo + productId;
        },
        apiDoInquiry: function() {
            return this.apiPath.doInquiry;
        },
        apiDoLogin: function() {
            return this.apiPath.doLogin;
        },
        apiInputMemberInfo: function() {
            return this.apiPath.inputMemberInfo;
        },
        apiConfirmMemberInfo: function() {
            return this.apiPath.confirmMemberInfo;
        },
        apiRegistMemberInfo: function() {
            return this.apiPath.registMemberInfo;
        },
        apiNewInfo: function(pageIndex) {
            return this.apiPath.newInfo + pageIndex;
        },
        apiCategory: function(categoryId, pageIndex) {
            return this.apiPath.category + categoryId + '&page=' + pageIndex;
        },
        apiDetail: function(productId) {
            return this.apiPath.detail + productId;
        },
        apiFulltextSearch: function() {
            return this.apiPath.fulltextSearch;
        },
        apiDoSearch: function() {
            return this.apiPath.doSearch;
        },
        apiInputShippingInfo: function() {
            return this.apiPath.inputShippingInfo;
        },
        apiInputShippingInfoNoMember: function() {
            return this.apiPath.inputShippingInfoNoMember;
        },
        apiConfirmShippingInfo: function() {
            return this.apiPath.confirmShippingInfo;
        },
        apiDoOrder: function() {
            return this.apiPath.doOrder;
        },
        apiTradeLaw: function() {
            return this.apiPath.tradeLaw;
        },
        apiOutputToken: function() {
            return this.apiPath.token;
        }
    }
}